### PR TITLE
fix(pipelines): support passing decoder model + tokenizer

### PR DESCRIPTION
**Pull request by dacorvo**
When passing explicitly a neuron model to a pipeline, we check the model class. This modifies the check to accept not only NeuronBaseModel but also NeuronModelForCausalLM.

Update: cherry-picked @glegendre01 modifications to github workflows to reactivate INF1 CI.

### CODE REVIEW

File: optimum/neuron/pipelines/transformers/base.py
Comment (michaelbenayoun) [inline optimum/neuron/pipelines/transformers/base.py:126]: QQ: why is `NeuronModelForCausalLM` not a sublass of `NeuronBseModel`?

Comment (dacorvo) [reply optimum/neuron/pipelines/transformers/base.py:126]: Because `NeuronBaseModel` is based on JIT models and implements the corresponding conversion logic. `NeuronModelForCausalLM` is a subclass of  `NeuronDecoderModel` that uses `transformers-neuronx` models instead.
We could refactor to add a common class to both though, since the latest subclasses of `NeuronBaseModel` are overriding pretty much all its methods now.

Comment (JingyaHuang) [reply optimum/neuron/pipelines/transformers/base.py:126]: Wouldn't it make more sense to use `NeuronDecoderModel` here?

**Review by JingyaHuang — APPROVED** [review]
LGTM, thanks for the fix!

File: optimum/neuron/pipelines/transformers/base.py
Comment (michaelbenayoun) [reply optimum/neuron/pipelines/transformers/base.py:126]: I guess it would be good to refactor it if possible. But also it's not pressing and can be postponed to when we have more bandwidth!

Comment (dacorvo) [reply optimum/neuron/pipelines/transformers/base.py:126]: I will wait until @JingyaHuang 's pull-request on T5 is merged as it is a new subclass, and any change I make before that to the base class will result in nightmarish conflicts.

**Pull request merged**
